foundationdb/src/api.rs

```diff
@@ -13,22 +13,230 @@
+                let handle = thread::spawn(move || {
+                    error::eval(unsafe { fdb_sys::fdb_run_network() }).expect("network start")
+                });
+                *self = GlobalState::Running {
+                    api_version: *api_version,
+                    handle,
+                    count: 1,
+                };
+                Ok(NetworkAutoStop { global: true })
+            }
+            GlobalState::Uninitialized => unreachable!(),
+            GlobalState::NetworkSetup { .. } => panic!("boot called on already setup network"),
+            GlobalState::Running { .. } => panic!("boot called on already running network"),
+            GlobalState::Stopped => panic!("the fdb network was stopped and can't be restarted"),
+        }
+    }
+    pub(crate) fn start(&mut self) -> FdbResult<Option<NetworkGuard>> {
+        match self {
+            GlobalState::Uninitialized | GlobalState::ApiVersionSelected { .. } => {
+                self.setup(true)?;
+                self.start()
+            }
+            GlobalState::NetworkSetup {
+                api_version,
+                managed: true,
+            } => {
+                let handle = thread::spawn(move || {
+                    error::eval(unsafe { fdb_sys::fdb_run_network() }).expect("network start")
+                });
+                // tests run in parallel and there is no "main" to boot and guard the network
+                // in this case, we add an implicit guard that will be dropped when the program exits
+                let count = if cfg!(feature="global_guard") {
+                    unsafe extern "C" {
+                        fn atexit(cb: extern "C" fn()) -> i32;
+                    }
+                    println!("GlobalState::start: register atexit hook");
+                    unsafe { atexit(GlobalState::exit) };
+                    2
+                } else {
+                    1
+                };
+                *self = GlobalState::Running {
+                    api_version: *api_version,
+                    handle,
+                    count,
+                };
+                Ok(Some(NetworkGuard { global: false }))
+            }
+            GlobalState::NetworkSetup { managed: false, .. } => {
+                println!("GlobalState::start: unmanaged");
+                // if not managed, the user is reponsible for starting/stopping the network
+                Ok(None)
+            }
+            GlobalState::Running { count, .. } => {
+                *count += 1;
+                println!("GlobalState::start: {count} guards alive");
+                Ok(Some(NetworkGuard { global: false }))
+            }
+            GlobalState::Stopped => panic!("the fdb network was stopped and can't be restarted"),
+        }
+    }
+    fn stop(&mut self) -> Option<FdbResult<()>> {
+        match self {
+            // stop is private and can only be called by NetworkGuard::drop
```

Comment:
GlobalState is a very generic name for this. How about something like FdbStateMachine or NetworkLifecycle or the like.